pkg/k8s/pod_utils.go

```diff
@@ -43,3 +44,45 @@ func LookupContainerPortAndName(pod *corev1.Pod, port intstr.IntOrString, protoc
 	}
 	return 0, "", errors.Errorf("unable to find port %s on pod %s", port.String(), NamespacedName(pod))
 }
+
+// StripDownPodTransformFunc is a transform function that strips down pod to reduce memory usage.
+// see details in [stripDownPodObject].
+func StripDownPodTransformFunc(obj interface{}) (interface{}, error) {
+	if pod, ok := obj.(*corev1.Pod); ok {
+		return stripDownPodObject(pod), nil
+	}
+	return obj, nil
```

Comment:
Seems we never return any error? If yes, can we remove error from return types? Or should we return error when converting objects fails?

Reply:
will remove the redundant nil.